processor/servicegraphprocessor/README.md

```diff
@@ -0,0 +1,110 @@
+
+Each emitted metrics series have the client and server label corresponding with the service doing the request and the service receiving the request.
+
+```
+traces_service_graph_request_total{client="app", server="db", connection_type="database"} 20
+```
+
+TLDR: The processor will try to find spans belonging to requests as seen from the client and the server and will create a metric representing an edge in the graph.
+
+## Metrics
+
+The following metrics are emitted by the processor:
+
+| Metric                                      | Type      | Labels                          | Description                                                  |
+|---------------------------------------------|-----------|---------------------------------|--------------------------------------------------------------|
+| traces_service_graph_request_total          | Counter   | client, server, connection_type | Total count of requests between two nodes                    |
+| traces_service_graph_request_failed_total   | Counter   | client, server, connection_type | Total count of failed requests between two nodes             |
+| traces_service_graph_request_server_seconds | Histogram | client, server, connection_type | Time for a request between two nodes as seen from the server |
+| traces_service_graph_request_client_seconds | Histogram | client, server, connection_type | Time for a request between two nodes as seen from the client |
+| traces_service_graph_unpaired_spans_total   | Counter   | client, server, connection_type | Total count of unpaired spans                                |
+| traces_service_graph_dropped_spans_total    | Counter   | client, server, connection_type | Total count of dropped spans                                 |
+
+Duration is measured both from the client and the server sides.
+
+Possible values for `connection_type`: unset, `messaging_system`, or `database`.
+
+Additional labels can be included using the `dimensions` configuration option.
+
+Since the service graph processor has to process both sides of an edge,
+it needs to process all spans of a trace to function properly.
+If spans of a trace are spread out over multiple instances, spans are not paired up reliably.
+
+## Visualization
+
+Service graph metrics are natively supported by Grafana since v9.0.4.
+To run it, configure a Tempo data source's 'Service Graphs' by linking to the Prometheus backend where metrics are being sent:
+
+```yaml
+apiVersion: 1
+datasources:
+  # Prometheus backend where metrics are sent
+  - name: Prometheus
+    type: prometheus
+    uid: prometheus
+    url: <prometheus-url>
+    jsonData:
+        httpMethod: GET
+    version: 1
+  - name: Tempo
+    type: tempo
+    uid: tempo
+    url: <tempo-url>
+    jsonData:
+      httpMethod: GET
+      serviceMap:
+        datasourceUid: 'prometheus'
+    version: 1
+```
+
+## Example configuration
+
+```yaml
+processors:
+  servicegraphs:
```

Comment:
Having now gone through the PR, I see that the readme is quite outdated. Could you give it a second pass? For instance, the processor needs a reference to a metrics exporter, which is not mentioned here. Instead, the readme says that metrics are generated in Prometheus format.

Reply:
You're right, it was very updated. Fixed in 60cfaf7119db2023323328c0cef25e08c6e9afdf, thanks.

Reply:
You're right. I've updated it with all the suggestions.